core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaOpenScreenTranslator.java

```diff
@@ -46,6 +51,14 @@ public void translate(GeyserSession session, ClientboundOpenScreenPacket packet)
 
         InventoryTranslator newTranslator = InventoryTranslator.inventoryTranslator(packet.getType());
         Inventory openInventory = session.getOpenInventory();
+
+        // Hack: ViaVersion translates the old (pre 1.20) smithing table to a furnace (does not work for Bedrock). We can detect this and translate it back to a smithing table.
+        if (session.isOldSmithingTable() &&
+            packet.getType() == ContainerType.FURNACE &&
+            packet.getTitle().equals(SMITHING_TABLE_COMPONENT)) {
+            newTranslator = new OldSmithingTableTranslator();
+        }
```

Comment:
Comment is longer than the if statement - I would just put the if-condition all on one line :p